inception-workload-dynamic/src/main/java/de/tudarmstadt/ukp/inception/workload/dynamic/support/AnnotationQueueOverviewDataProvider.java

```diff
@@ -300,50 +321,100 @@ public long getInProgressAmountForDocument(
     {
-        //Create an empty document
-        SourceDocument document = null;
-        Random r = new Random();
 
-        while (document == null)
-        {
-            int i = r.nextInt(data.size() - 1);
-            //If the random chosen document wont surpass the amount of default number combining
-            // "inProgress" for the document + "finished" amount for the document + 1
-            if ((getInProgressAmountForDocument(data.
-                get(i)) +
-                getFinishedAmountForDocument(data.
-                    get(i)))
-                + 1 <= defaultAnnotations)
-            {
-                //If that was not the case, assign this document
-                document = data.get(r.nextInt(data.size() - 1));
+
+        //First check for a documents that is in Progress for the user, and return this
+        for (int i = 0; i < allAnnotationDocuments.size(); i++) {
+            //NULL CHECK due to listAnnotableDocuments from webanno
+            if (allAnnotationDocuments.get(i) != null &&
+                allAnnotationDocuments.get(i).getState().equals(IN_PROGRESS)
+                && !allAnnotationDocuments.get(i).getName().equals(aCurrentAnno.getName())) {
+                //Found a document in progress
+                for (SourceDocument doc: data) {
+                    if (doc.getName().equals(allAnnotationDocuments.get(i).getName())) {
+                        return doc;
+                    }
+                }
+            }
+        }
+
+        //Nothing in Progress, so now return a random document, or create a new annotation document
+
+
+        //Get a random document
+        List<SourceDocument> srcList = entityManager.
+            createQuery(
+                "FROM SourceDocument " +
+                "WHERE project = :project " +
+                "ORDER BY rand()", SourceDocument.class)
+            .setParameter("project", aAnnotationPageBase.getModelObject().getProject())
+            .getResultList();
+        for (SourceDocument src: srcList) {
+            //Look if there is a corresponding Annotation document
+            //If not, create a new one
+            AnnotationDocument anno = entityManager.
+                createQuery(
+                    "FROM AnnotationDocument " +
+                    "WHERE name = :sourceName " +
+                    "AND user = :currentUser ", AnnotationDocument.class)
+                .setParameter("sourceName", src.getName())
+                .setParameter("currentUser", aAnnotationPageBase.
+                    getModelObject().getUser().getUsername())
+                .getResultList().stream().findFirst().orElse(null);
+
+            if (anno == null) {
+                //No annotation document for this source document
+                //for the current user, create a new Annotation document for the user
+                AnnotationDocument annotationDocument = new AnnotationDocument();
```

Comment:
Why do you still need to create new annotationdocuments here? We discussed this should not happen but instead you should use an adequate database query.

Reply:
Yes, this is why I call webannos createAnnotationDocument() a few lines below, which uses the new AnnotationDocument(). I do the creation exact the same way as it is done in webanno.
However, I can of course create my own query.

Reply:
If at some point we change the way how the annotation document is initialized, it will easily run out of sync with your code. Thus, I would prefer we avoid this risk by not introducing an additional point outside the WebAnno core code where annotation documents are created and saved.

Reply:
@christophscheunemann Did you address this issue?

Reply:
@jcklie Yes, I use now a query as requested to create the document. No more use of "new AnnotationDocument()" together with webannos createAnnotationsDocument()

Reply:
I can still see the code in the diff: 

https://github.com/inception-project/inception/pull/1727/files#diff-9e4ec6eba3c32eb8b97315524e282f71R363-R387